### PR TITLE
Move assert_allclose from chainer.gradient_check to chainer.testing

### PR DESCRIPTION
**Pull request by okuta**

### CODE REVIEW

**Comment by unnonouno** [comment]
I feel this PR causes a lot of conflicts with existing PRs. I want to merge this PR after almost all PRs are merged.

**Comment by okuta** [comment]
I split PR #1302.
Could you review this?

File: chainer/gradient_check.py [hunk 1]
```diff
@@ -73,14 +75,11 @@ def assert_allclose(x, y, atol=1e-5, rtol=1e-4, verbose=True):
...
-    x = cuda.to_cpu(utils.force_array(x))
-    y = cuda.to_cpu(utils.force_array(y))
-    try:
-        numpy.testing.assert_allclose(
-            x, y, atol=atol, rtol=rtol, verbose=verbose)
-    except Exception:
-        print('error:', numpy.abs(x - y).max())
-        raise
+    warnings.warn(
+        'chainer.gradient_check.assert_assert_allclose is deprecated.'
```
Comment (unnonouno) [inline chainer/gradient_check.py]: `assert_allclose`

File: chainer/gradient_check.py [hunk 2]
```diff
@@ -73,14 +75,11 @@ def assert_allclose(x, y, atol=1e-5, rtol=1e-4, verbose=True):
...
+        'Use chainer.testing.assert_assert_allclose instead.',
```
Comment (unnonouno) [inline chainer/gradient_check.py]: `assert_allclose`

**Comment by unnonouno** [comment]
LGTM except comments.

**Comment by unnonouno** [comment]
Can you rebase or merge the mater branch?

**Comment by unnonouno** [comment]
LGTM

**Pull request merged**